sharedresource/v1alpha1/types_share.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The OpenShift authors.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Share is the Schema for the shares API
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
```

Comment:
What controls whether or not this is installed?  Compatibility level 4 can be installed by default for non-user, non-cluster-admin APIs, but this reads like a cluster-admin API in which case it should not be installed by default.

Reply:
@deads2k the decision we received was that the driver/controller for this API would *always* be installed by OCP via the cluster storage operator, as one of the CSI drivers it manages.

Short term, while still in "tech preview" mode, @adambkaplan has https://github.com/openshift/api/pull/982 up for adding this to the tech preview set, so that CSO will only install that if tech preview is turned on for install.

Reply:
> @deads2k the decision we received was that the driver/controller for this API would _always_ be installed by OCP via the cluster storage operator, as one of the CSI drivers it manages.

You cannot install a techpreview API by default.  Installing the CRD by default means that you have to support upgrades.  To support upgrades, you need to be able to migrate data, which means your schema here has very restrictive upgrade constraints.

Your operator can create and manage this CRD definition.

Reply:
> Your operator can create and manage this CRD definition.

I think this is where we are going. The cluster storage operator will only install this CRD if the tech preview feature set is enabled or the right feature gate is enabled via the custom feature gate option. In either of these cases the cluster is not upgradable.

If the feature gate is disabled, the CRD is not installed.